app/javascript/src/Users/components/PermissionsForm.jsx

```diff
@@ -0,0 +1,109 @@
+// @flow
+
+import 'raf/polyfill'
+import 'core-js/es6/map'
+import 'core-js/es6/set'
+import 'core-js/es6/array'
+
+import React from 'react'
+
+import { Inputs } from 'Users/components/Inputs'
+import { RoleInput } from 'Users/components/RoleInput'
+import { UserRole } from 'Users/components/UserRole'
+import { FeatureAccessInput } from 'Users/components/FeatureAccessInput'
+import { FeatureAccess } from 'Users/components/FeatureAccess'
+import { ServiceFeatureAccess } from 'Users/components/ServiceFeatureAccess'
+import { ServiceAccessList } from 'Users/components/ServiceAccessList'
+import { ServiceAccess } from 'Users/components/ServiceAccess'
+
+import { isServicePermissionsGranted, getFeatureName, toggleElementInCollection } from 'Users/utils'
+import { createReactWrapper } from 'utilities/createReactWrapper'
+
+import type { Role, Feature } from 'Users/types'
+import type { Service } from 'Types'
+
+type Props = {
+  initialState: any,
+  features: Feature[],
+  services: Service[]
+}
+
+type State = {
+  role: Role,
+  checkedFeatures: Array<Feature | 'services'>,
+  checkedServicesIds: number[]
+}
+
+class PermissionsForm extends React.Component<Props, State> {
+  state = {
+    role: this.props.initialState.role || 'admin',
```

Comment:
we are passing the initialState as props? Is there a way to craft the initial state from the props? we also will be reducing the coupling.

Reply:
The state is being crafted from the props, it just happens the props is called `initialState` as [it was `state` before](https://github.com/3scale/porta/blob/master/app/views/provider/admin/account/users/_form.html.erb#L61). Do you think we should rename the props?